src/app/features/category/category.component.ts

```diff
@@ -1,12 +1,80 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { CategoryService } from '../../core/category-service/category.service';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
 
 @Component({
-  selector: 'srp-category-page',
+  selector: 'srp-category.component',
   standalone: true,
-  imports: [],
+  imports: [CommonModule , FormsModule],
   templateUrl: './category.component.html',
   styleUrl: './category.component.scss'
 })
-export class CategoryComponent {
+export class categoryComponent implements OnInit {
+  categories!: any[];
+  newCategoryName: string = '';
+  editCategoryId: string | null = null; 
+  editedCategoryName: string = ''; 
 
+  constructor(private categoryService: CategoryService) { }
+
+  ngOnInit(): void {
+    this.loadCategories();
+  }
+
+  loadCategories(): void {
+    this.categoryService.getCategories().subscribe(categories => {
```

Comment:
Don't subscribe for categories. instead we can use Observable stream. 
Read more about how to use async pipe